### ISSUE TITLE
Regression in 4.0

### ISSUE THREAD

**Issue by anlambert** [davidjbradshaw/iframe-resizer#671]
Since the 4.0 release, the `autoResize` option seems not honored anymore. 
More precisely, the iframe size gets no more adjusted to its content when the browser window gets resized.

Below is the log I obtain with the 4.0 version when resizing the browser window:
```
[iFrameSizer][Host page: null] Ignored: [object Object] iframeResizer.js:134:20
Navigated to http://localhost:5004/
[WDS] Disconnected! client:190:5
[iFrameSizer][Host page: null] Ignored: [object Object] iframeResizer.js:134:20
Navigated to http://localhost:5004/
[iFrameSizer][Host page: iFrameResizer0] Added missing iframe ID: iFrameResizer0 (http://localhost:5004/coverage/) iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] IFrame scrolling disabled for iFrameResizer0 iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] [init] Sending msg to iframe[iFrameResizer0] (iFrameResizer0:8:false:true:32:true:true:null:taggedElement:null:null:0:false:parent:scroll) targetOrigin: http://localhost:5004 iframeResizer.js:134:20
[iFrameSizer][Host page: null] Ignored: [object Object] iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] Added missing iframe ID: iFrameResizer0 (http://localhost:5004/coverage/) iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] IFrame scrolling disabled for iFrameResizer0 iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] [init] Sending msg to iframe[iFrameResizer0] (iFrameResizer0:8:false:true:32:true:true:null:taggedElement:null:null:0:false:parent:scroll) targetOrigin: http://localhost:5004 iframeResizer.js:134:20
[iFrameSizer][Host page: null] Ignored: [object Object] iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] [iFrame.onload] Sending msg to iframe[iFrameResizer0] (iFrameResizer0:8:false:true:32:true:true:null:taggedElement:null:null:0:false:parent:scroll) targetOrigin: http://localhost:5004 iframeResizer.js:134:20
[iFrameSizer][iFrameResizer0] Initialising iFrame (http://localhost:5004/coverage/) iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] TargetOrigin for parent set to: * iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] height calculation method set to "taggedElement" iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] HTML & body height set to "auto" iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Enable public methods iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Animation Start iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Animation Iteration iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Animation End iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Input iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Mouse Up iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Mouse Down iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Orientation Change iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Print iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Ready State Change iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Touch Start iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Touch End iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Touch Cancel iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Transition Start iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Transition Iteration iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Add event listener: Transition End iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Create body MutationObserver iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] In page linking not enabled iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Parsed 1 HTML elements iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Element position calculated in 1ms iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Trigger event lock on iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] Sending message to host page (iFrameResizer0:172:1110:init) iframeResizer.contentWindow.js:169:7
[iFrameSizer][Host page: iFrameResizer0] Received: [iFrameSizer]iFrameResizer0:172:1110:init iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] Checking connection is from: http://localhost:5004 iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] Checking height is in range 0-Infinity iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] Checking width is in range 0-Infinity iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] IFrame (iFrameResizer0) height set to 172px iframeResizer.js:134:20
[iFrameSizer][iFrameResizer0] Trigger event lock off iframeResizer.contentWindow.js:169:7
[iFrameSizer][iFrameResizer0] -- iframeResizer.contentWindow.js:169:7
[iFrameSizer][Host page: window] Trigger event: resize iframeResizer.js:134:20 
```

While the same operation gives me the following log with version 3.6.5:
```
Navigated to http://localhost:5004/
[WDS] Disconnected! client:190:5
[iFrameSizer][Host page: null] Ignored: [object Object] iframeResizer.js:134:20
[iFrameSizer][Host page: iFrameResizer0] Added missing iframe ID: iFrameResizer0 (http://localhost:5004/coverage/) iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] IFrame scrolling disabled for iFrameResizer0 iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] [init] Sending msg to iframe[iFrameResizer0] (iFrameResizer0:8:false:true:32:true:true:null:taggedElement:null:null:0:false:parent:scroll) targetOrigin: http://localhost:5004 iframeResizer.js:143:20
[iFrameSizer][Host page: null] Ignored: [object Object] iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] [iFrame.onload] Sending msg to iframe[iFrameResizer0] (iFrameResizer0:8:false:true:32:true:true:null:taggedElement:null:null:0:false:parent:scroll) targetOrigin: http://localhost:5004 iframeResizer.js:143:20
[iFrameSizer][iFrameResizer0] Initialising iFrame (http://localhost:5004/coverage/) iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] TargetOrigin for parent set to: * iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] height calculation method set to "taggedElement" iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] HTML & body height set to "auto" iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Enable public methods iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Animation Start iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Animation Iteration iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Animation End iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Input iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Mouse Up iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Mouse Down iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Orientation Change iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Print iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Ready State Change iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Touch Start iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Touch End iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Touch Cancel iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Transition Start iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Transition Iteration iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Add event listener: Transition End iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Create body MutationObserver iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] In page linking not enabled iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Parsed 1 HTML elements iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Element position calculated in 1ms iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Trigger event lock on iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Sending message to host page (iFrameResizer0:172:1110:init) iframeResizer.contentWindow.js:187:7
[iFrameSizer][Host page: iFrameResizer0] Received: [iFrameSizer]iFrameResizer0:172:1110:init iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] Checking connection is from: http://localhost:5004 iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] Checking height is in range 0-Infinity iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] Checking width is in range 0-Infinity iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] IFrame (iFrameResizer0) height set to 172px iframeResizer.js:143:20
[iFrameSizer][iFrameResizer0] Trigger event lock off iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] -- iframeResizer.contentWindow.js:187:7
[iFrameSizer][Host page: window] Trigger event: resize iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] [Window resize] Sending msg to iframe[iFrameResizer0] (resize) targetOrigin: http://localhost:5004 iframeResizer.js:143:20
[iFrameSizer][iFrameResizer0] Trigger event: Parent window requested size check iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Parsed 1 HTML elements iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Element position calculated in 0ms iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Trigger event lock on iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Sending message to host page (iFrameResizer0:860:510:resizeParent) iframeResizer.contentWindow.js:187:7
[iFrameSizer][Host page: iFrameResizer0] Received: [iFrameSizer]iFrameResizer0:860:510:resizeParent iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] Checking connection is from: http://localhost:5004 iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] Checking height is in range 0-Infinity iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] Checking width is in range 0-Infinity iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] Requesting animation frame iframeResizer.js:143:20
[iFrameSizer][Host page: window] Trigger event: resize iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] IFrame (iFrameResizer0) height set to 860px iframeResizer.js:143:20
[iFrameSizer][Host page: iFrameResizer0] [Window resize] Sending msg to iframe[iFrameResizer0] (resize) targetOrigin: http://localhost:5004 iframeResizer.js:143:20
[iFrameSizer][iFrameResizer0] Trigger event: Parent window requested size check iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Parsed 1 HTML elements iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Element position calculated in 0ms iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] No change in size detected iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] Trigger event lock off iframeResizer.contentWindow.js:187:7
[iFrameSizer][iFrameResizer0] -- iframeResizer.contentWindow.js:187:7 
```

It seems some resize handlers get no more executed.

**Comment by davidjbradshaw** [comment]
No sure how I missed that, will take a look

**Comment by anlambert** [comment]
If you need more details on my iframe-resizer setup:

  - the page holding the iframe to resize: https://forge.softwareheritage.org/source/swh-web/browse/master/swh/web/templates/homepage.html$115

   - the iframe page: https://forge.softwareheritage.org/source/swh-web/browse/master/swh/web/templates/coverage.html$51

**Comment by davidjbradshaw** [comment]
Fixed in v4.0.2, thanks for raising the issue.

**Comment by anlambert** [comment]
Thanks for the fixes, I confirm that it works now. Nevertheless I stumbled across a new issue in 4.0.3, see #674.
